docs/virtual_machine_template_guide.adoc

```diff
@@ -0,0 +1,209 @@
+= VirtualMachine Templates
+
+VirtualMachine templates are openshift templates (link:https://docs.openshift.org/latest/dev_guide/templates.html[Openshift Templates]) +
+that describe VirtualMachine specifications which produce the following template objects: +
+**One** OfflineVirtualMachine and **0-N** PersistentVolumeClaims. +
+The following document describes the process of creating a Virtual machine template which will be managed by ManageIQ.
+
+== Writing Templates
+You can define new VirtualMachine templates to make it easy to recreate all relevant objects (OfflineVirtualMachine and PersistentVolumeClaim). The template will define the objects it creates along with some metadata to guide the creation of those objects.
+
+[source,yaml]
+----
+apiVersion: kubevirt.io/v1alpha1
+kind: Template
+metadata:
+  name: RHEL-7 ❶
+  annotations:
+  labels:
+    miq.github.io/kubevirt-os: rhel-7 ❷
+    miq.github.io/kubevirt-is-vm-template: True ❸
+objects:
+- apiVersion: kubevirt.io/v1alpha1
+  kind: OfflineVirtualMachine ❹
+  metadata:
+    name: ${NAME} ➎
+  spec:
+    template:
+      metadata:
+        labels:
+          my: label
+      spec:
+        domain: 
+          resources: ➏
+            requests:
+              memory: ${MEMORY} 
+          cpu: ❼
+            cores: ${CPU}
+          devices:
+            disks: ❽
+              name: disk0
+              volumeName: mypvc 
+      volumes: ➒
+       - name: mypvc
+         presistentVolumeClaim:
+           claimName: rhel-7-pvc-${NAME}
+- apiVersion: kubevirt.io/v1alpha1
+  kind: PersistentVolumeClaim ➓
+  metadata:
+    name: rhel-7-pvc-${NAME} ⓫
+    #FIXME: add an annotation to specify which volume this pvc should be based on
```

Comment:
This is not enough information to create pvc. Kubernetes is complaining about:

> * spec.accessModes: Required value: at least 1 access mode is required
> * spec.resources[storage]: Required value

Adding following spec solved the issue for me:
```yaml
   spec:
    accessModes:
    - ReadWriteOnce
    resources:
      requests:
        storage: 1Gi
```

Reply:
@yanirq can you include this?

Reply:
@pkliczewski  @fabiand  done